crates/transaction-pool/src/pool/best_helper.rs

```diff
@@ -0,0 +1,61 @@
+#![allow(dead_code)]
+use std::sync::Arc;
+
+use crate::{pool::best::BestTransactions, TransactionOrdering, ValidPoolTransaction};
+
+pub(crate) enum InvalidPatterns {
+    // prevent state change leakage by private transactions
+    PrivateTransaction,
+}
+
+pub(crate) struct BestTransactionsHelper<T: TransactionOrdering> {
+    pub(crate) best: BestTransactions<T>,
+    pub(crate) invalid_patterns: Vec<InvalidPatterns>,
+}
```

Comment:
this is not quite what we want here because this limits us to predefined patterns.

what we want is something like the std Filter

https://doc.rust-lang.org/stable/src/core/iter/adapters/filter.rs.html#18